src/attack.c

```diff
@@ -832,7 +860,7 @@ check_kb_access (int soc)
   kb_t kb;
 
   rc = kb_new (&kb, prefs_get ("kb_location"));
-  if (rc)
+  if (rc && is_otp_scan())
```

Comment:
Also, is this needed ? Given that you already call is_otp_scan() in error_message_to_client().